ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -136,15 +138,18 @@ internal constructor(
         .distinctUntilChanged()
   }
 
-  private fun toLocationOfInterestFeatures(
+  private suspend fun toLocationOfInterestFeatures(
     locationsOfInterest: Set<LocationOfInterest>
   ): Set<Feature> = // TODO: Add support for polylines similar to mapPins.
   locationsOfInterest
       .map {
+        val pendingSubmissions =
+          localSubmissionStore.getPendingSubmissionCountByLocationOfInterestId(it.id)
+        val submissionCount = it.submissionCount + pendingSubmissions
```

Comment:
Combining data from different data sources is generally the responsibility of the repository layer. Can we move this there?

You may want to also subtract deleted submissions now so we don't forget to do it later.

Reply:
Done!